crates/bevy_transform/src/components/children.rs

```diff
@@ -1,18 +1,22 @@
 use bevy_ecs::{Entity, MapEntities};
 use bevy_reflect::{Reflect, ReflectComponent, ReflectMapEntities};
+use bevy_utils::HashMap;
 use smallvec::SmallVec;
 use std::ops::Deref;
 
-#[derive(Default, Clone, Debug, Reflect)]
+#[derive(Default, Clone, Debug, Reflect, PartialEq, Eq)]
 #[reflect(Component, MapEntities)]
-pub struct Children(pub(crate) SmallVec<[Entity; 8]>);
+pub struct Children {
+    order: SmallVec<[Entity; 8]>,
```

Comment:
IMO we should consider pulling in `smolset` to do this for us, rather than worrying about enforcing uniqueness ourselves. I've been poking at this space for storing input maps too.